generators/csharp/sdk/src/endpoint/http/HttpEndpointGenerator.ts

```diff
@@ -312,4 +309,347 @@ export class HttpEndpointGenerator extends AbstractEndpointGenerator {
+                    offset: (pagination) => {
+                        const offsetType = this.context.csharpTypeMapper.convert({
+                            reference: pagination.page.property.valueType
+                        });
+                        // use specified type or fallback to int
+                        const stepType = pagination.step
+                            ? this.context.csharpTypeMapper.convert({
+                                  reference: pagination.step?.property.valueType
+                              })
+                            : csharp.Type.object();
+                        const offsetPagerClassReference = this.context.getOffsetPagerClassReference({
+                            requestType: requestParam.type,
+                            requestOptionsType,
+                            responseType: unpagedEndpointResponseType,
+                            offsetType,
+                            stepType,
+                            itemType
+                        });
+                        writer.write("var pager = new ");
+                        writer.writeNode(offsetPagerClassReference);
+                        writer.write("(");
+                        writer.indent();
+                        writer.writeLine(`${requestParam.name},`);
+                        writer.writeLine(`${optionsParamName},`);
+                        writer.writeLine(`${unpagedEndpointMethodName},`);
+                        writer.writeLine(`request => ${this.nullableDotGet("request", pagination.page)} ?? 0,`);
+
+                        writer.writeLine("(request, offset) => {");
+                        writer.indent();
+                        this.initializeNestedObjects(writer, "request", pagination.page);
+                        writer.writeTextStatement(`${this.dotGet("request", pagination.page)} = offset`);
+                        writer.dedent();
+                        writer.writeLine("},");
+
+                        if (pagination.step) {
+                            writer.writeLine(`request => ${this.nullableDotGet("request", pagination.step)},`);
+                        } else {
+                            writer.writeLine("null,");
+                        }
+                        writer.writeLine(
+                            `response => ${this.nullableDotGet("response", pagination.results)}?.ToList(),`
+                        );
+                        if (pagination.hasNextPage) {
+                            writer.writeLine(`response => ${this.nullableDotGet("response", pagination.hasNextPage)}`);
+                        } else {
+                            writer.writeLine("null");
+                        }
+                        writer.dedent();
+                        writer.writeTextStatement(")");
+                        writer.writeTextStatement("return pager");
+                    },
+                    cursor: (pagination) => {
```

Comment:
nit: Could we put all of this in a separate function (e.g. `generateCursorPagerMethod`).